plugins/color_panel/macos/FLEColorPanelPlugin.mm

```diff
@@ -45,26 +45,30 @@ - (instancetype)initWithChannel:(FLEMethodChannel*)channel {
  * panel channel.
  */
 - (void)handleMethodCall:(FLEMethodCall *)call result:(FLEMethodResult)result {
-  BOOL handled = YES;
+  BOOL methodImplemented = YES;
   if ([call.methodName isEqualToString:@(plugins_color_panel::kShowColorPanelMethod)]) {
     if ([call.arguments isKindOfClass:[NSDictionary class]]) {
       BOOL showAlpha =
           [[call.arguments valueForKey:@(plugins_color_panel::kColorPanelShowAlpha)] boolValue];
       [self showColorPanelWithAlpha:showAlpha];
     } else {
-      NSLog(@"Malformed call for %@. Expected an NSDictionary but got %@",
-            @(plugins_color_panel::kShowColorPanelMethod),
-            NSStringFromClass([call.arguments class]));
-      handled = NO;
+      NSString *errorString =
+          [NSString stringWithFormat:@"Malformed call for %@. Expected an NSDictionary but got %@",
+                                     @(plugins_color_panel::kShowColorPanelMethod),
+                                     NSStringFromClass([call.arguments class])];
+      result([[FLEMethodError alloc] initWithCode:@"error"
+                                          message:@"Bad arguments"
+                                          details:errorString]);
+      return;
     }
   } else if ([call.methodName isEqualToString:@(plugins_color_panel::kHideColorPanelMethod)]) {
     [self hideColorPanel];
   } else {
-    handled = NO;
+    methodImplemented = NO;
   }
   // Send an immediate empty success message for handled messages, since the actual color data
   // will be provided in follow-up messages.
-  result(handled ? nil : FLEMethodNotImplemented);
+  result(methodImplemented ? nil : FLEMethodNotImplemented);
```

Comment:
Mixing early return and return-at-end-with-local-variable in the same method is very confusing. Let's remove `methodImplemented` entirely, and have the end of each branch call `result` with the appropriate value.